hack/smoke.sh

```diff
@@ -0,0 +1,104 @@
+#!/bin/bash
+#
+# Usage:
+#
+# To use this script for smoke testing, you'll need to extract a valid user
+# token from the relevant URL.  Note that tokens aren't portable, i.e. a token
+# from console.redhat.com won't work on console.dev.redhat.com and vice versa.
+
+function check_command {
+    OUTPUT="$(command -v "${1}")"
+    if [[ "${OUTPUT}" = "" ]]; then
+        echo "${1} required for smoke tests!"
+        exit 1
+    fi
+}
+
+function display_help {
+    echo "help for $(basename "${0}")"
+    echo "-e    --environment   Sets the environment to run smoke tests on"
+    echo "-h    --help          Print this help message"
+    echo "-t    --token         Specifies the JWT token to use"
+}
+
+function match_environment {
+    case "${1}" in
+        staging-internal)
+            echo "Internal staging is not yet supported!"
+            exit 1
+            ;;
+        production-internal)
+            echo "Internal production is not yet supported!"
+            exit 1
+            ;;
+        staging-external)
+            CLUSTER_URL="https://console.dev.redhat.com"
+            ;;
+        production-external)
+            CLUSTER_URL="https://console.redhat.com"
+            ;;
+        *)
+            CLUSTER_URL="${1}"
+    esac
+}
+
+while [[ $# -gt 0 ]]; do
+    case "${1}" in
+        -h | --help)
+            display_help
+            exit 0
+            ;;
+        -t | --token)
+            shift
+            TOKEN="${1}"
+            shift
+            ;;
+        -e | --environment)
+            shift
+            match_environment "${1}"
+            shift
+            ;;
+        -u | --user)
+            shift
+            USERNAME="${1}"
+            shift
+            ;;
+        *)
+            echo "Unrecognized command ${1}!"
+            display_help
+            exit 1
+    esac
+done
+
+# TODO(sadlerap): can we extract the username from the token?
```

Comment:
it will work for most of the cases, but for some users may not work. If the username is too long or it starts with a forbidden prefix, KubeSaw modifies it. I'd say we can default it to token's `username` but the user should be able to change it.

Something like the following may do the trick  (not tested):
```bash
if [[ -z "${USERNAME}" ]]; then
    USERNAME=$(jq -R 'split(".") | .[1] | @base64d | fromjson | .username' <<< "${TOKEN}")
fi
```

Reply:
With the knowledge that it might be modified by kubesaw, I'm thinking it might be a better idea to not try to do any username extraction and instead have the user always provide it.

Reply:
With 53cf9f7, I've removed this `TODO`.